NickvisionCavalier.GNOME/Controls/ImageItem.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Threading.Tasks;
+
+namespace NickvisionCavalier.GNOME.Controls;
+
+/// <summary>
+/// Image item in background image selector
+/// </summary>
+public class ImageItem : Gtk.Overlay
+{
+    private readonly Gtk.Picture _picture;
+    private readonly Gtk.Button _removeButton;
+    private readonly int _index;
+    
+    /// <summary>
+    /// Occurs when remove button was clicked
+    /// </summary>
+    public event Action<int>? OnRemoveImage;
+    
+    /// <summary>
+    /// Construct ImageItem
+    /// </summary>
+    public ImageItem(string path, int index)
+    {
+        _index = index;
+        SetMarginTop(2);
+        SetMarginStart(2);
+        SetMarginEnd(2);
+        SetMarginBottom(2);
+        _picture = Gtk.Picture.New();
+        _picture.AddCssClass("cavalier-image");
+        _picture.SetSizeRequest(160, 120);
+        _picture.SetContentFit(Gtk.ContentFit.Cover);
+        Task.Run(() =>
+        {
+            var texture = Gdk.Texture.NewFromFilename(path); // This is thread-safe
+            GLib.Functions.IdleAdd(0, () =>
+            {
+                _picture.SetPaintable(texture);
```

Comment:
I believe Texture is disposable...please call `texture.Dispose()` after this line.

Reply:
thanks, that seems to fix memory leak that I couldn't find